straxen/contexts.py

```diff
@@ -218,17 +219,100 @@ def xenonnt_led(**kwargs):
+                       _allow_diverging_settings=False,
+                       **kwargs):
+    """
+    Context with already pre-initialized corrections values for CMT.
+    Sets correction values according cmt_run_id and specified version.
+    Settings can be overwritten via cmt_overwrite or fax config file.
+
+    :param output_folder: Output folder for strax data.
+    :param cmt_run_id: Run id which should be used to load corrections.
+    :param cmt_version: Global version for corrections to be loaded.
+    :param cmt_overwrite: Dictionary to overwrite CMT settings. Keys
+        must be valid CMT option keys.
+    :param fax_config: Fax config file to use.
+    :param overwrite_from_fax_file: If true overwrites CMT settings
+        from fax file.
+    :param _allow_diverging_settings: Disable check for matching CMT
+        settings in context and fax file settings. Experts use only.
+    :param kwargs: Additional kwargs taken by strax.Context.
+
+    :return: strax.Context instance
+    """
     import wfsim
     st = strax.Context(
         storage=strax.DataDirectory(output_folder),
         config=dict(detector='XENONnT',
-                    fax_config='fax_config_nt_design.json',
+                    fax_config=fax_config,
                     check_raw_record_overlaps=True,
-                    **straxen.contexts.xnt_simulation_config,
+                    **straxen.contexts.xnt_common_config,
                     ),
-        **straxen.contexts.xnt_common_opts)
+        **straxen.contexts.xnt_common_opts,  **kwargs)
     st.register(wfsim.RawRecordsFromFaxNT)
+
+    st.apply_cmt_version(f'global_{cmt_version}')
+    if not cmt_run_id:
+        raise ValueError('You have to specify a run_id which should be used to initialize '
+                         'the corrections.')
+
+    cmt_options = straxen.get_corrections.get_cmt_options(st)
+
+    p = st.get_single_plugin(cmt_run_id, 'peak_positions')
+    default_reconstruction_algorithm = p.config['default_reconstruction_algorithm']
+    
+    def _depends_on_pos_algo(cmt_setting):
+        """
+        Checks if CMT setting depends on a position reconstruction
+        algorithm.
+        """
+        config_name, setting, nT = cmt_setting
+        if config_name in ('s1_xyz_map', 'fdc_map'):
+            cmt_setting = (config_name + f'_{default_reconstruction_algorithm}', setting, nT)
+        return cmt_setting
+
+    for option_key, cmt_setting in cmt_options.items():
+        cmt_setting = _depends_on_pos_algo(cmt_setting)
+        config_name, _, _ = _depends_on_pos_algo(cmt_setting)
+        config_value = straxen.get_correction_from_cmt(cmt_run_id, cmt_setting)
+        if isinstance(config_value, np.ndarray):
+            # We have to convert the arrays into a list to write the metadata
+            # numpy arrays are not json serializable.
+            config_value = config_value.tolist()
+
+        cmt_options[option_key] = (config_name + '_constant',
```

Comment:
Hi @WenzDaniel , sorry about the late comment. I still don't really get why we can not just change everything in cmt_setting to `('MC', cmt_runid, cmt_setting)`? Isn't it still a valid value to convert to a hash?

Reply:
I understood from Diego that this cause an issue with the lineage. If this still works there would not be a need for a new context.